dbms/src/Storages/MergeTree/MergeTreeDataWriter.cpp

```diff
@@ -146,8 +146,12 @@ MergeTreeData::MutableDataPartPtr MergeTreeDataWriter::writeTempPart(BlockWithDa
             ProfileEvents::increment(ProfileEvents::MergeTreeDataWriterBlocksAlreadySorted);
     }
 
+    auto compression_method = data.context.chooseCompressionMethod(
+        data.getTotalActiveSizeInBytes(),
+        static_cast<double>(data.getTotalActiveSizeInBytes()) / data.getTotalCompressedSize());
```

Comment:
Honestly, I started (my local version of code) works with single compression method (both `min_part_size` and `min_part_size_ratio` set to 0 in config) but then prepared a pull request so I thought if it is likely to be accepted it should respect existing configuration of "compression" tag so I thought I have to use chooseCompressionMethod as it should be. If you are OK to use `chooseCompressionMethod(0,0)` I can add it.